// @staticsai/statics/models/components/accountdto.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountDto$ = exports.AccountDto$outboundSchema = exports.AccountDto$inboundSchema = exports.AccountDtoStatus$ = exports.AccountDtoStatus$outboundSchema = exports.AccountDtoStatus$inboundSchema = exports.AccountDtoStatus = void 0;
const z = __importStar(require("zod"));
/**
 * Status of the account
 */
exports.AccountDtoStatus = {
    Active: "ACTIVE",
    Closed: "CLOSED",
    NeedsRefresh: "NEEDS_REFRESH",
};
/** @internal */
exports.AccountDtoStatus$inboundSchema = z.nativeEnum(exports.AccountDtoStatus);
/** @internal */
exports.AccountDtoStatus$outboundSchema = exports.AccountDtoStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountDtoStatus$;
(function (AccountDtoStatus$) {
    /** @deprecated use `AccountDtoStatus$inboundSchema` instead. */
    AccountDtoStatus$.inboundSchema = exports.AccountDtoStatus$inboundSchema;
    /** @deprecated use `AccountDtoStatus$outboundSchema` instead. */
    AccountDtoStatus$.outboundSchema = exports.AccountDtoStatus$outboundSchema;
})(AccountDtoStatus$ || (exports.AccountDtoStatus$ = AccountDtoStatus$ = {}));
/** @internal */
exports.AccountDto$inboundSchema = z.object({
    id: z.string(),
    orgId: z.string(),
    appId: z.string(),
    status: exports.AccountDtoStatus$inboundSchema,
    createdAt: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    updatedAt: z.string().datetime({ offset: true }).transform(v => new Date(v)),
});
/** @internal */
exports.AccountDto$outboundSchema = z.object({
    id: z.string(),
    orgId: z.string(),
    appId: z.string(),
    status: exports.AccountDtoStatus$outboundSchema,
    createdAt: z.date().transform(v => v.toISOString()),
    updatedAt: z.date().transform(v => v.toISOString()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountDto$;
(function (AccountDto$) {
    /** @deprecated use `AccountDto$inboundSchema` instead. */
    AccountDto$.inboundSchema = exports.AccountDto$inboundSchema;
    /** @deprecated use `AccountDto$outboundSchema` instead. */
    AccountDto$.outboundSchema = exports.AccountDto$outboundSchema;
})(AccountDto$ || (exports.AccountDto$ = AccountDto$ = {}));
