// @staticsai/statics/models/components/linkdto.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkDto$ = exports.LinkDto$outboundSchema = exports.LinkDto$inboundSchema = exports.Status$ = exports.Status$outboundSchema = exports.Status$inboundSchema = exports.Status = void 0;
const z = __importStar(require("zod"));
/**
 * The current status of the link
 */
exports.Status = {
    Uninitialized: "UNINITIALIZED",
    Connecting: "CONNECTING",
    Ready: "READY",
    Closed: "CLOSED",
};
/** @internal */
exports.Status$inboundSchema = z
    .nativeEnum(exports.Status);
/** @internal */
exports.Status$outboundSchema = exports.Status$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Status$;
(function (Status$) {
    /** @deprecated use `Status$inboundSchema` instead. */
    Status$.inboundSchema = exports.Status$inboundSchema;
    /** @deprecated use `Status$outboundSchema` instead. */
    Status$.outboundSchema = exports.Status$outboundSchema;
})(Status$ || (exports.Status$ = Status$ = {}));
/** @internal */
exports.LinkDto$inboundSchema = z.object({
    id: z.string(),
    status: exports.Status$inboundSchema,
    linkToken: z.string(),
    accountId: z.string().optional(),
});
/** @internal */
exports.LinkDto$outboundSchema = z.object({
    id: z.string(),
    status: exports.Status$outboundSchema,
    linkToken: z.string(),
    accountId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LinkDto$;
(function (LinkDto$) {
    /** @deprecated use `LinkDto$inboundSchema` instead. */
    LinkDto$.inboundSchema = exports.LinkDto$inboundSchema;
    /** @deprecated use `LinkDto$outboundSchema` instead. */
    LinkDto$.outboundSchema = exports.LinkDto$outboundSchema;
})(LinkDto$ || (exports.LinkDto$ = LinkDto$ = {}));
