// @staticsai/statics/src/models/errors/getaccount.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Account not found
 */
export type GetAccountResponseBodyData = {
  statusCode: number;
  message: string;
  error?: string | undefined;
};

/**
 * Account not found
 */
export class GetAccountResponseBody extends Error {
  statusCode: number;
  error?: string | undefined;

  /** The original data that was passed to this error instance. */
  data$: GetAccountResponseBodyData;

  constructor(err: GetAccountResponseBodyData) {
    const message = "message" in err && typeof err.message === "string"
      ? err.message
      : `API error occurred: ${JSON.stringify(err)}`;
    super(message);
    this.data$ = err;

    this.statusCode = err.statusCode;
    if (err.error != null) this.error = err.error;

    this.name = "GetAccountResponseBody";
  }
}

/** @internal */
export const GetAccountResponseBody$inboundSchema: z.ZodType<
  GetAccountResponseBody,
  z.ZodTypeDef,
  unknown
> = z.object({
  statusCode: z.number(),
  message: z.string(),
  error: z.string().optional(),
})
  .transform((v) => {
    return new GetAccountResponseBody(v);
  });

/** @internal */
export type GetAccountResponseBody$Outbound = {
  statusCode: number;
  message: string;
  error?: string | undefined;
};

/** @internal */
export const GetAccountResponseBody$outboundSchema: z.ZodType<
  GetAccountResponseBody$Outbound,
  z.ZodTypeDef,
  GetAccountResponseBody
> = z.instanceof(GetAccountResponseBody)
  .transform(v => v.data$)
  .pipe(z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
  }));

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetAccountResponseBody$ {
  /** @deprecated use `GetAccountResponseBody$inboundSchema` instead. */
  export const inboundSchema = GetAccountResponseBody$inboundSchema;
  /** @deprecated use `GetAccountResponseBody$outboundSchema` instead. */
  export const outboundSchema = GetAccountResponseBody$outboundSchema;
  /** @deprecated use `GetAccountResponseBody$Outbound` instead. */
  export type Outbound = GetAccountResponseBody$Outbound;
}
