// @staticsai/statics/src/models/operations/getaccount.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

export type GetAccountRequest = {
  /**
   * The ID of the account
   */
  accountId: string;
};

/** @internal */
export const GetAccountRequest$inboundSchema: z.ZodType<
  GetAccountRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  accountId: z.string(),
});

/** @internal */
export type GetAccountRequest$Outbound = {
  accountId: string;
};

/** @internal */
export const GetAccountRequest$outboundSchema: z.ZodType<
  GetAccountRequest$Outbound,
  z.ZodTypeDef,
  GetAccountRequest
> = z.object({
  accountId: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetAccountRequest$ {
  /** @deprecated use `GetAccountRequest$inboundSchema` instead. */
  export const inboundSchema = GetAccountRequest$inboundSchema;
  /** @deprecated use `GetAccountRequest$outboundSchema` instead. */
  export const outboundSchema = GetAccountRequest$outboundSchema;
  /** @deprecated use `GetAccountRequest$Outbound` instead. */
  export type Outbound = GetAccountRequest$Outbound;
}
