// @staticsai/statics/src/models/operations/getlink.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

export type GetLinkRequest = {
  /**
   * The ID of the link that was returned by creation endpoint earlier.
   */
  linkId?: any | undefined;
};

/** @internal */
export const GetLinkRequest$inboundSchema: z.ZodType<
  GetLinkRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  linkId: z.any().optional(),
});

/** @internal */
export type GetLinkRequest$Outbound = {
  linkId?: any | undefined;
};

/** @internal */
export const GetLinkRequest$outboundSchema: z.ZodType<
  GetLinkRequest$Outbound,
  z.ZodTypeDef,
  GetLinkRequest
> = z.object({
  linkId: z.any().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetLinkRequest$ {
  /** @deprecated use `GetLinkRequest$inboundSchema` instead. */
  export const inboundSchema = GetLinkRequest$inboundSchema;
  /** @deprecated use `GetLinkRequest$outboundSchema` instead. */
  export const outboundSchema = GetLinkRequest$outboundSchema;
  /** @deprecated use `GetLinkRequest$Outbound` instead. */
  export type Outbound = GetLinkRequest$Outbound;
}
