// @staticsai/statics/src/models/components/createproxydto.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

export type CreateProxyDto = {
  /**
   * ID of the account you are trying to access
   */
  accountId: string;
};

/** @internal */
export const CreateProxyDto$inboundSchema: z.ZodType<
  CreateProxyDto,
  z.ZodTypeDef,
  unknown
> = z.object({
  accountId: z.string(),
});

/** @internal */
export type CreateProxyDto$Outbound = {
  accountId: string;
};

/** @internal */
export const CreateProxyDto$outboundSchema: z.ZodType<
  CreateProxyDto$Outbound,
  z.ZodTypeDef,
  CreateProxyDto
> = z.object({
  accountId: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CreateProxyDto$ {
  /** @deprecated use `CreateProxyDto$inboundSchema` instead. */
  export const inboundSchema = CreateProxyDto$inboundSchema;
  /** @deprecated use `CreateProxyDto$outboundSchema` instead. */
  export const outboundSchema = CreateProxyDto$outboundSchema;
  /** @deprecated use `CreateProxyDto$Outbound` instead. */
  export type Outbound = CreateProxyDto$Outbound;
}
