// @staticsai/statics/src/models/components/createlinkdto.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

export type CreateLinkDto = {
  /**
   * ID of the application that you are trying to link
   */
  appId: string;
};

/** @internal */
export const CreateLinkDto$inboundSchema: z.ZodType<
  CreateLinkDto,
  z.ZodTypeDef,
  unknown
> = z.object({
  appId: z.string(),
});

/** @internal */
export type CreateLinkDto$Outbound = {
  appId: string;
};

/** @internal */
export const CreateLinkDto$outboundSchema: z.ZodType<
  CreateLinkDto$Outbound,
  z.ZodTypeDef,
  CreateLinkDto
> = z.object({
  appId: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CreateLinkDto$ {
  /** @deprecated use `CreateLinkDto$inboundSchema` instead. */
  export const inboundSchema = CreateLinkDto$inboundSchema;
  /** @deprecated use `CreateLinkDto$outboundSchema` instead. */
  export const outboundSchema = CreateLinkDto$outboundSchema;
  /** @deprecated use `CreateLinkDto$Outbound` instead. */
  export type Outbound = CreateLinkDto$Outbound;
}
