// @staticsai/statics/models/errors/createproxy.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateProxyResponseBody$ = exports.CreateProxyResponseBody$outboundSchema = exports.CreateProxyResponseBody$inboundSchema = exports.CreateProxyProxyResponseBody$ = exports.CreateProxyProxyResponseBody$outboundSchema = exports.CreateProxyProxyResponseBody$inboundSchema = exports.CreateProxyResponseBody = exports.CreateProxyProxyResponseBody = void 0;
const z = __importStar(require("zod"));
/**
 * The account either does not exist or is not in your tenant
 */
class CreateProxyProxyResponseBody extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.statusCode = err.statusCode;
        if (err.error != null)
            this.error = err.error;
        this.name = "CreateProxyProxyResponseBody";
    }
}
exports.CreateProxyProxyResponseBody = CreateProxyProxyResponseBody;
/**
 * The account is not in a valid state for proxy
 */
class CreateProxyResponseBody extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.statusCode = err.statusCode;
        if (err.error != null)
            this.error = err.error;
        this.name = "CreateProxyResponseBody";
    }
}
exports.CreateProxyResponseBody = CreateProxyResponseBody;
/** @internal */
exports.CreateProxyProxyResponseBody$inboundSchema = z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
})
    .transform((v) => {
    return new CreateProxyProxyResponseBody(v);
});
/** @internal */
exports.CreateProxyProxyResponseBody$outboundSchema = z.instanceof(CreateProxyProxyResponseBody)
    .transform(v => v.data$)
    .pipe(z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateProxyProxyResponseBody$;
(function (CreateProxyProxyResponseBody$) {
    /** @deprecated use `CreateProxyProxyResponseBody$inboundSchema` instead. */
    CreateProxyProxyResponseBody$.inboundSchema = exports.CreateProxyProxyResponseBody$inboundSchema;
    /** @deprecated use `CreateProxyProxyResponseBody$outboundSchema` instead. */
    CreateProxyProxyResponseBody$.outboundSchema = exports.CreateProxyProxyResponseBody$outboundSchema;
})(CreateProxyProxyResponseBody$ || (exports.CreateProxyProxyResponseBody$ = CreateProxyProxyResponseBody$ = {}));
/** @internal */
exports.CreateProxyResponseBody$inboundSchema = z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
})
    .transform((v) => {
    return new CreateProxyResponseBody(v);
});
/** @internal */
exports.CreateProxyResponseBody$outboundSchema = z.instanceof(CreateProxyResponseBody)
    .transform(v => v.data$)
    .pipe(z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateProxyResponseBody$;
(function (CreateProxyResponseBody$) {
    /** @deprecated use `CreateProxyResponseBody$inboundSchema` instead. */
    CreateProxyResponseBody$.inboundSchema = exports.CreateProxyResponseBody$inboundSchema;
    /** @deprecated use `CreateProxyResponseBody$outboundSchema` instead. */
    CreateProxyResponseBody$.outboundSchema = exports.CreateProxyResponseBody$outboundSchema;
})(CreateProxyResponseBody$ || (exports.CreateProxyResponseBody$ = CreateProxyResponseBody$ = {}));
