// @staticsai/statics/models/errors/createlink.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateLinkResponseBody$ = exports.CreateLinkResponseBody$outboundSchema = exports.CreateLinkResponseBody$inboundSchema = exports.CreateLinkLinkResponseBody$ = exports.CreateLinkLinkResponseBody$outboundSchema = exports.CreateLinkLinkResponseBody$inboundSchema = exports.CreateLinkResponseBody = exports.CreateLinkLinkResponseBody = void 0;
const z = __importStar(require("zod"));
/**
 * The application requested was not found
 */
class CreateLinkLinkResponseBody extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.statusCode = err.statusCode;
        if (err.error != null)
            this.error = err.error;
        this.name = "CreateLinkLinkResponseBody";
    }
}
exports.CreateLinkLinkResponseBody = CreateLinkLinkResponseBody;
class CreateLinkResponseBody extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.statusCode = err.statusCode;
        if (err.error != null)
            this.error = err.error;
        this.name = "CreateLinkResponseBody";
    }
}
exports.CreateLinkResponseBody = CreateLinkResponseBody;
/** @internal */
exports.CreateLinkLinkResponseBody$inboundSchema = z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
})
    .transform((v) => {
    return new CreateLinkLinkResponseBody(v);
});
/** @internal */
exports.CreateLinkLinkResponseBody$outboundSchema = z.instanceof(CreateLinkLinkResponseBody)
    .transform(v => v.data$)
    .pipe(z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateLinkLinkResponseBody$;
(function (CreateLinkLinkResponseBody$) {
    /** @deprecated use `CreateLinkLinkResponseBody$inboundSchema` instead. */
    CreateLinkLinkResponseBody$.inboundSchema = exports.CreateLinkLinkResponseBody$inboundSchema;
    /** @deprecated use `CreateLinkLinkResponseBody$outboundSchema` instead. */
    CreateLinkLinkResponseBody$.outboundSchema = exports.CreateLinkLinkResponseBody$outboundSchema;
})(CreateLinkLinkResponseBody$ || (exports.CreateLinkLinkResponseBody$ = CreateLinkLinkResponseBody$ = {}));
/** @internal */
exports.CreateLinkResponseBody$inboundSchema = z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
})
    .transform((v) => {
    return new CreateLinkResponseBody(v);
});
/** @internal */
exports.CreateLinkResponseBody$outboundSchema = z.instanceof(CreateLinkResponseBody)
    .transform(v => v.data$)
    .pipe(z.object({
    statusCode: z.number(),
    message: z.string(),
    error: z.string().optional(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateLinkResponseBody$;
(function (CreateLinkResponseBody$) {
    /** @deprecated use `CreateLinkResponseBody$inboundSchema` instead. */
    CreateLinkResponseBody$.inboundSchema = exports.CreateLinkResponseBody$inboundSchema;
    /** @deprecated use `CreateLinkResponseBody$outboundSchema` instead. */
    CreateLinkResponseBody$.outboundSchema = exports.CreateLinkResponseBody$outboundSchema;
})(CreateLinkResponseBody$ || (exports.CreateLinkResponseBody$ = CreateLinkResponseBody$ = {}));
